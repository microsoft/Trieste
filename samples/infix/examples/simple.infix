x = 5;
print "x" x;
y = 2 - 1;
print "1 + 10" 1 + 10;
