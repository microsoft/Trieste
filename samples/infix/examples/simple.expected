//! parse_only
//! --enable-tuples parse_only
//! --enable-tuples --tuples-require-parens parse_only
//! --enable-tuples --tuples-require-parens --use-parser-tuples parse_only
(top
  {}
  (infix-calculation
    {
      x = infix-assign
      y = infix-assign}
    (infix-assign
      (infix-ident 1:x)
      (infix-expression
        (infix-int 1:5)))
    (infix-output
      (infix-string 3:"x")
      (infix-expression
        (infix-ref
          (infix-ident 1:x))))
    (infix-assign
      (infix-ident 1:y)
      (infix-expression
        (infix-subtract
          (infix-expression
            (infix-int 1:2))
          (infix-expression
            (infix-int 1:1)))))
    (infix-output
      (infix-string 8:"1 + 10")
      (infix-expression
        (infix-add
          (infix-expression
            (infix-int 1:1))
          (infix-expression
            (infix-int 2:10)))))))
